code/game/machinery/doors/airlock.dm

```diff
@@ -1691,6 +1691,25 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 	if(exposed_temperature > (T0C + heat_resistance))
 		take_damage(round(exposed_volume / 100), BURN, 0, 0)
 
+/obj/machinery/door/airlock/wrench_act(mob/living/user, obj/item/tool)
+	if(!locked)
+		return
```

Comment:
I think you might need to return TRUE from these interactions to prevent it from just smacking the airlock after the interaction is done.

Also, if this interaction doesn't work with a regular wrench, I'd move the istype to be a guard clause (early return) somewhere up here, possibly with a message.

Reply:
It's more of an attack chain thing -- iirc if you don't return TRUE here, it'll continue down the attack chain and try attackby, before ultimately just getting to "you hit x with y". If it currently works as you'd expect it to, then that's fine and you can ignore me here.

For the second point, it looks like you have it now with the
```dm
if(!istype(tool, /obj/item/wrench/bolter))
		return
```

and since you have this, you can remove the check down below to see if we do have the right tool.

Reply:
> It's more of an attack chain thing -- iirc if you don't return TRUE here, it'll continue down the attack chain and try attackby, before ultimately just getting to "you hit x with y". If it currently works as you'd expect it to, then that's fine and you can ignore me here.

more so miss understood/miss interpreted what you meant. i've added your suggestion